### PR TITLE
Add initial version of homepage and login page

### PR DESCRIPTION
**Pull request by hht-hannah**
f53b4dcf959c4854adf43f63595bbdd
39aeee5601e3a49e7c72d0dc8365921
c6f083c6787c70a6da1f68d1ba76565
b9e039fc65230a2891de60ad2f6acd6
Here are some screenshots.

### CODE REVIEW

File: flutter/Graminsta/lib/auth.dart [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        });
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+        body: new SafeArea(
+      child: new SingleChildScrollView(
```
Comment (Time1ess) [inline flutter/Graminsta/lib/auth.dart]: Try to format with `dartfmt`

File: flutter/Graminsta/lib/auth.dart [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:Graminsta/login.dart';
+
+class AuthPage extends StatefulWidget {
+  AuthPage({Key key}) : super(key: key);
+
+  @override
+  _AuthPageState createState() => new _AuthPageState();
```
Comment (Time1ess) [inline flutter/Graminsta/lib/auth.dart]: Not sure if I'm right, please double check. As part of the Dart 2, `new` is optional. and Effective Dart suggests not to use that.

See also: https://dart.dev/guides/language/effective-dart/usage#dont-use-new

Comment (Time1ess) [reply flutter/Graminsta/lib/auth.dart]: Ditto for all.

File: flutter/Graminsta/lib/login.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+
+//登陆界面
```
Comment (Time1ess) [inline flutter/Graminsta/lib/login.dart]: Nit: try to comment in English.

And the class name is self-explainable, so you can omit the documentation.

File: flutter/Graminsta/lib/login.dart [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            textColor: Colors.white,
+            child: new Text('Login'),
+            onPressed: () {},
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget LoginForm() {
```
Comment (Time1ess) [inline flutter/Graminsta/lib/login.dart]: You can inline this function or make it private.

File: flutter/Graminsta/lib/auth.dart [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        body: new SafeArea(
+      child: new SingleChildScrollView(
+          child: new Container(
+              height: MediaQuery.of(context).size.height,
+              width: MediaQuery.of(context).size.width,
+              child: new Column(
+                mainAxisSize: MainAxisSize.max,
+                children: <Widget>[
+                  new SizedBox(
+                    height: 75,
```
Comment (Time1ess) [inline flutter/Graminsta/lib/auth.dart]: For constants, let's create a `spec` directory in the project root and put constants in it.

```
Graminsta/
Graminsta/spec/lib/spacing.dart
Graminsta/spec/lib/sizing.dart   // For sizes like font sizes, box heights
```

For example:

```
// spacing.dart
...
const mediumVerticalElementSpacing = 16;
const largexxx = 24;
const xLargexxx = 48;
```

File: flutter/Graminsta/lib/auth.dart [hunk 6]
```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
```
Comment (Time1ess) [inline flutter/Graminsta/lib/auth.dart]: Put auth.dart and login.dart under `auth` directory.

```
Graminsta/auth/lib/auth.dart
...
```

**Review by Time1ess — CHANGES_REQUESTED** [review]
Please try to provide some screenshots for your pages, otherwise it's hard for reviewers to figure out the real UI of these pages.

**Comment by shiyi001** [comment]
How about submitting a PR `fix .gitignore` separately?

**Comment by isabella0428** [comment]
To add screenshots, click Attach files by dragging & dropping, selecting or pasting them in the lower part of comment.
WechatIMG558

**Comment by isabella0428** [comment]
To add screenshots, click Attach files by dragging & dropping, selecting or pasting them in the lower part of comment.
WechatIMG558

File: flutter/Graminsta/lib/auth.dart (hunk 5 above)
Comment (hht-hannah) [reply flutter/Graminsta/lib/auth.dart]: I create a directory under the lib dir if it's ok.
```
Graminsta/lib/spec/spacing.dart
Graminsta/lib/spec/sizing.dart
```
Sorry that I couldn't figure out how to import the package if the directory is under the root dir.

Comment (shiyi001) [reply flutter/Graminsta/lib/auth.dart]: try `import 'package:Graminsta/spec/spacing.dart';`

**Comment by isabella0428** [comment]
Maybe move the login& register page content a little bit lower? It seems that the content is too high.

File: flutter/Graminsta/lib/auth/login.dart [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    return Container(
+      padding: const EdgeInsets.all(20),
+      child: Column(
+        children: <Widget>[
+          _LoginForm(),
+          MaterialButton(
+            color: Colors.blue,
+            textColor: Colors.white,
+            child: Text('Login'),
+            onPressed: () {},
```
Comment (Time1ess) [inline flutter/Graminsta/lib/auth/login.dart]: I think you can omit this field if you don't have callback yet.

**Review by Time1ess — DISMISSED** [review]

File: flutter/Graminsta/lib/auth.dart (hunk 5 above)
Comment (hht-hannah) [reply flutter/Graminsta/lib/auth.dart]: The import statement works fine now. But it won't work if I move the file into the root directory.

File: flutter/Graminsta/lib/auth/login.dart (hunk 7 above)
Comment (hht-hannah) [reply flutter/Graminsta/lib/auth/login.dart]: It seems that the field onPressed is required for material button.

**Review by Time1ess — APPROVED** [review]

**Review by shiyi001 — APPROVED** [review]
Great job!

**Review by ArianaGu — APPROVED** [review]
Great job!

**Pull request merged**